harness/determined/_training.py

```diff
@@ -0,0 +1,113 @@
+import enum
+import logging
+from typing import Any, Dict, List, Optional
+
+import determined as det
+from determined.common.api import errors
+from determined.experimental import client
+
+log = logging.getLogger("generic")
+
+
+class EarlyExitReason(enum.Enum):
+    INVALID_HP = "EXITED_REASON_INVALID_HP"
+    # This is generally unnecessary; just exit early.
+    USER_REQUESTED_STOP = "EXITED_REASON_USER_REQUESTED_STOP"
+
+
+class Training:
+    """
+    Some training-related REST API wrappers.
+    """
+
+    def __init__(self, session: client.Session, trial_id: int, run_id: int, exp_id: int) -> None:
```

Comment:
what is `run_id` here?  is it what we decided to call `allocation`?

Reply:
Yes it is.  And it's already called `allocation_id` in the push feature branch, but I haven't rebased those changes to here just yet.